assets/src/hooks/useNotificationsReducer.ts

```diff
@@ -183,9 +195,82 @@ const persistToggledNotificationReadState = (
   ])
 }
 
-export const useNotificationsReducer = (): [State, Dispatch] => {
+type InitialNotificationData = { initial_notifications: NotificationData[] }
+
+type InitialNotifications = { type: "initial"; payload: Notification[] }
+type NewNotification = { type: "new"; payload: Notification }
+
+type ReceivedNotifications = NewNotification | InitialNotifications | null
+
+const parseNotifications = (
+  notificationData: NotificationData | InitialNotificationData
+): NewNotification | InitialNotifications => {
+  if ("initial_notifications" in notificationData) {
+    return {
+      type: "initial",
+      payload: (
+        notificationData as InitialNotificationData
+      ).initial_notifications.map(notificationFromData),
+    }
+  }
+  return {
+    type: "new",
+    payload: notificationFromData(notificationData as NotificationData),
+  }
+}
+
+const useNotifications = (socket: Socket | undefined, routeIds: RouteId[]) => {
+  return useChannel<ReceivedNotifications>({
+    socket,
+    topic: "notifications",
+    event: "notification",
+    parser: parseNotifications,
+    dependencies: routeIds,
+    loadingState: null,
+  })
+}
+
+export const useNotificationsReducer = (
+  isInitialLoad: boolean,
+  setIsInitialLoad: React.Dispatch<boolean>
+): [State, Dispatch] => {
+  const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
   const [state, dispatch] = useReducer(reducer, initialState)
 
+  const [{ routeTabs }] = useContext(StateDispatchContext)
+  const [routeIds, setRouteIds] = useState<RouteId[]>(
+    allOpenRouteIds(routeTabs)
+  )
+  const newRouteIds = allOpenRouteIds(routeTabs)
+
+  if (!equalByElements(routeIds, newRouteIds)) {
+    setRouteIds(newRouteIds)
+  }
+
+  const latestMessage: ReceivedNotifications = useNotifications(
+    socket,
+    routeIds
+  )
```

Comment:
**praise:** I really like this approach for coupling the different reducers together, I was sort of grasping at something like this idea but couldn't quite put it together in a way that worked.